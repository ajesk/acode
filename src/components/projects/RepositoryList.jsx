import React from 'react';
import Repository from './Repository';
import './RepositoryList.scss';

const RepositoryList = ({ repositories }) => {
  return (
    <div className="repository-list">
      <h2>Repositories</h2>

      {repositories.sort((a, b) => Date.parse(b.updated_at) - Date.parse(a.updated_at))
        .map(repo =>
          <Repository repo={repo} />
        )}
    </div>
  );
}

export default RepositoryList;
