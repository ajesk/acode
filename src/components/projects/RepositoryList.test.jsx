import React from 'react';
import { render } from '@testing-library/react';
import RepositoryList from './RepositoryList';

describe('repository list', () => {
  const repositories = [
    {
      id: 'ajdflkah392iauk',
      html_url: 'fackall',
      name: 'adsfn,vxmz,mnlknf dsfsdf',
      language: 'javer',
      updated_at: 'today'
    },{
      id: '222',
      html_url: '222',
      name: '2222',
      language: 'english',
      updated_at: '2222'
    }
];
  test('snapshot', () => {
    expect(render(<RepositoryList repositories={repositories} />)).toMatchSnapshot();
  });
});