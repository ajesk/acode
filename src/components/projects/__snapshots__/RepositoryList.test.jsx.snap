// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repository list snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="repository-list"
      >
        <h2>
          Repositories
        </h2>
        <div
          class="repo"
        >
          <a
            class="repo-link"
            href="fackall"
          >
            adsfn,vxmz,mnlknf dsfsdf
          </a>
          <div
            class="repo-details"
          >
            <div
              class="repo-left"
            >
              <div>
                Language:
              </div>
              <div>
                Last Updated:
              </div>
            </div>
            <div
              class="repo-right"
            >
              <div>
                javer
              </div>
              <div>
                today
              </div>
            </div>
          </div>
        </div>
        <div
          class="repo"
        >
          <a
            class="repo-link"
            href="222"
          >
            2222
          </a>
          <div
            class="repo-details"
          >
            <div
              class="repo-left"
            >
              <div>
                Language:
              </div>
              <div>
                Last Updated:
              </div>
            </div>
            <div
              class="repo-right"
            >
              <div>
                english
              </div>
              <div>
                2222
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="repository-list"
    >
      <h2>
        Repositories
      </h2>
      <div
        class="repo"
      >
        <a
          class="repo-link"
          href="fackall"
        >
          adsfn,vxmz,mnlknf dsfsdf
        </a>
        <div
          class="repo-details"
        >
          <div
            class="repo-left"
          >
            <div>
              Language:
            </div>
            <div>
              Last Updated:
            </div>
          </div>
          <div
            class="repo-right"
          >
            <div>
              javer
            </div>
            <div>
              today
            </div>
          </div>
        </div>
      </div>
      <div
        class="repo"
      >
        <a
          class="repo-link"
          href="222"
        >
          2222
        </a>
        <div
          class="repo-details"
        >
          <div
            class="repo-left"
          >
            <div>
              Language:
            </div>
            <div>
              Last Updated:
            </div>
          </div>
          <div
            class="repo-right"
          >
            <div>
              english
            </div>
            <div>
              2222
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
